packages/compass-aggregations/src/modules/pipeline-builder/text-editor-pipeline.ts

```diff
@@ -50,7 +49,7 @@ export type TextEditorState = {
   stageOperators: string[];
   syntaxErrors: PipelineParserError[];
   serverError: MongoServerError | null;
-  loading: boolean;
+  isLoading: boolean;
   previewDocs: Document[] | null;
```

Comment:
Feel free to keep as is, but that's not really what I suggested, what confuses me about this store structure is that you have now two slices that are running actions and keeping state for the same UI which makes it harder to reason about (e.g., what does it mean if both loading flags are true? what does it mean if both completed flags return true? both error? what if there is still an error in this slice but not one in another, what if it's the other way around?). You have a clear piece of the UI that is dedicated to output, it can be either preview results or output result, it can't be both, why not move all relevant state and actions into this new slice you added if you want to keep them separate?

Reply:
maybe i am missing something, but the two slices are completely independent. have their own state, which is consumed separately on UI (except for showing server error).

Reply:
They could be independent, but this is not how UI code is written. Here's an example of the broken state:

![image](https://user-images.githubusercontent.com/5036933/199499693-94ad4e6c-7976-4006-bbbb-cfc55bdcca1b.png)